### PR TITLE
feat: add ability to specify existing secret for Django SECRET_KEY

### PR DESCRIPTION
**Pull request by hazzadous**
Previously the behaviour was to look for an existing value for the
secret and, if one doesn't exist, to generate one automatically.

There wasn't a way to explicitly specify the key, which means that if
you want to point a new deployment at existing database resources,
you'll end up invalidating e.g. Django issued cookies.

## Type of change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by guidoiaquinti — APPROVED** [review]
Overall LGTM but what about removing the reference to Django? With that we could have a more generic `posthog-env`

**Comment by hazzadous** [comment]
> Overall LGTM but what about removing the reference to Django? With that we could have a more generic `posthog-env`

Yup wasn't sure about that either, will update.

File: charts/posthog/templates/migrate.job.yaml [hunk 1]
```diff
@@ -66,11 +66,8 @@ spec:
           value: {{ template "posthog.site.url" . }}
         - name: DEPLOYMENT
           value: {{ template "posthog.deploymentEnv" . }}
-        - name: SECRET_KEY
-          valueFrom:
-            secretKeyRef:
-              name: {{ template "posthog.fullname" . }}
-              key: posthog-secret
+        # Django env variables
```
Comment (guidoiaquinti) [inline charts/posthog/templates/migrate.job.yaml:69]:
```suggestion
  # PostHog app settings
```

File: charts/posthog/templates/secrets.yaml [hunk 2]
```diff
@@ -9,6 +9,7 @@ type: Opaque
 data:
   {{- if $previous }}
   posthog-secret: {{ index $previous.data "posthog-secret" }}
+  {{- else if .Values.posthogSecretKey.existingSecret }}
```
Comment (guidoiaquinti) [inline charts/posthog/templates/secrets.yaml:12]: is this empty?

File: charts/posthog/templates/web-deployment.yaml [hunk 3]
```diff
@@ -90,11 +90,8 @@ spec:
         - name: SECURE_COOKIES
           value: '0'
       {{- end }}
-        - name: SECRET_KEY
-          valueFrom:
-            secretKeyRef:
-              name: {{ template "posthog.fullname" . }}
-              key: posthog-secret
+        # Django env variables
```
Comment (guidoiaquinti) [inline charts/posthog/templates/web-deployment.yaml:93]:
```suggestion
  # PostHog app settings
```

**Pull request merged**